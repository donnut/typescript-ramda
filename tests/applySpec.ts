import R_applySpec = require('../ramda/dist/src/applySpec');

interface Name {
  firstname: string;
  lastname: string;
}

declare const name_spec: { [K in keyof Name]: (fullname: string) => Name[K] };

declare const fullname: string;

// @dts-jest:pass:snap
R_applySpec(name_spec);
// @dts-jest:pass:snap
R_applySpec(name_spec)(fullname);

// @dts-jest:pass:snap
R_applySpec<Name>(name_spec);
// @dts-jest:pass:snap
R_applySpec<Name>(name_spec)(fullname);
