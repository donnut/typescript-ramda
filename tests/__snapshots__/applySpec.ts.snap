// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`R.applySpec(name_spec) 1`] = `"(...args: any[]) => {}"`;

exports[`R.applySpec(name_spec)(fullname) 1`] = `"{}"`;

exports[`R.applySpec<Name>(name_spec) 1`] = `"(...args: any[]) => Name"`;

exports[`R.applySpec<Name>(name_spec)(fullname) 1`] = `"Name"`;
